### ISSUE TITLE
Invalid initial expressions in test_steady_state_basin_mms.py

### ISSUE THREAD

**Issue by wence-** [thetisproject/thetis#43]
In setup9:

```python
    grad_h_x = '''h0/(2.0*lx*sqrt(0.3*x[0]*x[0] + 0.2*x[1]*x[1]))*0.6*x[0] - 3.0*pi/lx*sin(pi*(3.0*x[0] + 1.0*x[1])/lx)'''
    grad_h_y = '''h0/(2.0*lx*sqrt(0.3*x[0]*x[0] + 0.2*x[1]*x[1]))*0.4*x[1] - 1.0*pi/lx*sin(pi*(3.0*x[0] + 1.0*x[1])/lx)'''
```

If `x` is the origin (entirely plausible).  Then we end up with:

```python
grad_h_x = h0/(2*lx*sqrt(0))*0.6*0 - ...
```

=> BOOM.

**Comment by wence-** [comment]
Notice that one might not always notice this.  Because although the physical mesh coordinate is exactly zero, the way we do the interpolation is to get fiat to evaluate the dual set at the finite element nodes.  This should be the identity matrix, but sometimes it is not *quite* the identity (you have some 1e-16 on the off diagonals).  As a result, in that case you don't get a NaN.

**Comment by tkarna** [comment]
Yes, thanks for tracing this down, we'll update the setups.

**Comment by wence-** [comment]
While here, I highly recommend switching from pasted C strings to interpolation via UFL expressions.

**Comment by tkarna** [comment]
fixed in #44
